pkg/common/resources/resources.go

```diff
@@ -993,3 +993,39 @@ func CalculateAbsUsedCapacity(capacity, used *Resource) *Resource {
 	}
 	return absResource
 }
+
+// DominantResourceType calculates the most used resource type based on the ratio of used compared to
+// the capacity. If a capacity type is set to 0 assume full usage.
+// Dominant type should be calculated with queue usage and capacity. Queue capacities should never
+// contain 0 values when there is a usage also, however in the root queue this could happen. If the
+// last node reporting that resource was removed but not everything has been updated.
+// immediately
+// Ignores resources types that are used but not defined in the capacity.
+func (r *Resource) DominantResourceType(capacity *Resource) string {
```

Comment:
I think we should treat zero capacity a bit differently. if a resource has 0 capacity and 0 usage, just ignore it. If it has > 0 usage, then treat it as 100%.

Reply:
done: both cap and usage 0 is 0%, cap 0 with usage > 0 is 100%, otherwise: usage / cap